builtin/logical/aws/path_roles.go

```diff
@@ -56,8 +56,11 @@ func pathRoles(b *backend) *framework.Path {
 			},
 
 			"policy_arns": &framework.FieldSchema{
-				Type:        framework.TypeCommaStringSlice,
-				Description: "ARNs of AWS policies to attach to IAM users. Only valid when credential_type is " + iamUserCred,
+				Type: framework.TypeCommaStringSlice,
+				Description: fmt.Sprintf(`ARNs of AWS policies. Behavior varies by credential_type. When credential_type is
+%s, then it will attach the specified policies to the generated IAM user.
+When credential_type is %s or %s, the policies will be passed as the
+PolicyArns parameter, acting as a filter on permissions available.`, iamUserCred, assumedRoleCred, federationTokenCred),
```

Comment:
I do still wonder if this should actually be a new, separate field due to the very different behavior this has depending on the role type. I am flexible though. Are there reasons you feel this is best?

Reply:
Consistency. The behavior of `policy_arns` field is essentially identical to the behavior of the `policy_document` field. Since there's only one `policy_document` field for all credential types, I would want there to be a single `policy_arns` field for all credential types. Otherwise, I think it would be even more confusing to break out `policy_arns` and not `policy_document`.

Now, I don't have strong opinions about whether the `policy_*` fields should be renamed based on `credential_type`, only that they should be consistent. If you want to break them up, that's fine, but let's do that in a separate PR (whether that's before this one or after) so that everything remains consistent.